pkg/provider/docker/config.go

```diff
@@ -345,7 +345,17 @@ func (p Provider) getIPAddress(ctx context.Context, container dockerData) string
 			logger.Warnf("Unable to get IP address for container %s : Failed to inspect container ID %s, error: %s", container.Name, connectedContainer, err)
 			return ""
 		}
-		return p.getIPAddress(ctx, parseContainer(containerInspected))
+
+		// Check connected container for traefik.docker.network, falling back to
+		// the network specified on the current container.
+		containerParsed := parseContainer(containerInspected)
+		extraConf, _ := p.getConfiguration(containerParsed)
+		if extraConf.Docker.Network == "" {
+			extraConf.Docker.Network = container.ExtraConf.Docker.Network
+		}
```

Comment:
I don't think this does anything useful. Whatever container you're looking at, I think its ExtraConf.Docker.Network always comes from p.Network anyway, so it's the same value everywhere.
Or am I missing anything?

Reply:
ah nevermind, my bad, I see now that it can be overridden by the specific label on the container.

Reply:
Yeah I'm not sure that's really what we want. I initially had labelled the service container expecting it to work, but in the course of doing this PR, it seemed to make more sense to label the container which owned the network stack.

What do you think? Is my approach of checking both reasonable enough?